sda-commons-starter/src/main/java/org/sdase/commons/starter/SdaPlatformBundle.java

```diff
@@ -0,0 +1,352 @@
+
+    @Override
+    public ConsumerTokenConfigBuilder<SdaPlatformConfiguration> usingSdaPlatformConfiguration() {
+      return usingCustomConfig(SdaPlatformConfiguration.class)
+          .withAuthConfigProvider(SdaPlatformConfiguration::getAuth)
+          .withCorsConfigProvider(SdaPlatformConfiguration::getCors);
+    }
+
+    @Override
+    public <T extends Configuration> AuthConfigProviderBuilder<T> usingCustomConfig(
+        Class<T> configurationClass) {
+      return new InitialBuilder<>();
+    }
+
+    // CustomConfigurationProviders and follow up configuration providers
+
+    @Override
+    public CorsConfigProviderBuilder<C> withoutAuthentication() {
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withAuthConfigProvider(
+        AuthConfigProvider<C> authConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withAnnotatedAuthorization();
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withOpaAuthorization(
+        AuthConfigProvider<C> authConfigProvider, OpaConfigProvider<C> opaConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withExternalAuthorization();
+      this.opaBundleBuilder = OpaBundle.builder().withOpaConfigProvider(opaConfigProvider);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withoutCorsSupport() {
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withCorsConfigProvider(
+        CorsConfigProvider<C> corsConfigProvider) {
+      this.corsBundleBuilder = CorsBundle.builder().withCorsConfigProvider(corsConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withoutConsumerTokenSupport() {
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withOptionalConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(true);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenRequiredConfigInitialBuilder<C> withRequiredConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(false);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withConsumerTokenConfigProvider(
+        ConsumerTokenConfigProvider<C> consumerTokenConfigProvider) {
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(consumerTokenConfigProvider);
+      return this;
+    }
```

Comment:
Open for discussion: Is there ever a need to make this configurable? I would also vote to remove it from the default `SdaPlatformConfiguration` class if it is not the default.

Reply:
I've searched in our organization and found some occurrences:  
[1](https://github.com/SDA-SE/edithcare-participation-service/blob/fa61f6d544d833373d895721b3b803ad60ff4245/config.yml#L21)
[2](https://github.com/SDA-SE/customer-service/blob/fea7b0fb679a51d2835b0621940fc91ea31ab0fc/src/integTest/resources/test-config.yml#L16)
[3](https://github.com/SDA-SE/billing-external/blob/15fe3b544b7606f3dcfd68903789609a6f55f893/billing-external-service/config.yaml#L10)
And it seems some others are using it as well. [search in our organization](https://github.com/search?l=YAML&q=org%3ASDA-SE+consumerToken&type=Code)

I would also vote to remove it from the `SdaPlatformConfiguration`.

Reply:
Please take a look at [PR-662](https://github.com/SDA-SE/sda-dropwizard-commons/pull/662)